token/program-2022/src/error.rs

```diff
@@ -156,9 +155,17 @@ pub enum TokenError {
             the associated `maximum_pending_balance_credit_counter`"
     )]
     MaximumPendingBalanceCreditCounterExceeded,
+
+    // 40
     /// The deposit amount for the confidential extension exceeds the maximum limit
     #[error("Deposit amount exceeds maximum limit")]
     MaximumDepositAmountExceeded,
+    /// Cpi Guard is enabled, and a program tried to take a protected action via cpi
+    #[error("Cpi Guard is enabled, and a program tried to take a protected action via cpi")]
+    CpiGuardEnabled,
+    /// Cpi Guard status cannot be changed in cpi
+    #[error("Cpi Guard status cannot be changed in cpi")]
+    CpiGuardSettingsLocked,
```

Comment:
i might replace the first error here with a different error for every single disallowed action. but im not sure if thats "too much"